--- a/src/fuxi-gmac-phy.c	2026-04-29 09:33:55.093504671 +0200
+++ b/src/fuxi-gmac-phy.c	2026-04-29 09:34:54.683563336 +0200
@@ -322,7 +322,9 @@
 static void fxgmac_phy_link_poll(unsigned long data)
 #endif
 {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6,15,0))
+    struct fxgmac_pdata *pdata = timer_container_of(pdata, t, expansion.phy_poll_tm);
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0))
     struct fxgmac_pdata *pdata = from_timer(pdata, t, expansion.phy_poll_tm);
 #else
     struct fxgmac_pdata *pdata = (struct fxgmac_pdata*)data;
@@ -350,14 +352,17 @@
 
 int fxgmac_phy_timer_init(struct fxgmac_pdata *pdata)
 {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6,15,0))
+    timer_setup(&pdata->expansion.phy_poll_tm, fxgmac_phy_link_poll, 0);
+    pdata->expansion.phy_poll_tm.expires = jiffies + HZ / 2;
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0))
     init_timer_key(&pdata->expansion.phy_poll_tm, NULL, 0, "fuxi_phy_link_update_timer", NULL);
+    pdata->expansion.phy_poll_tm.expires = jiffies + HZ / 2;
+    pdata->expansion.phy_poll_tm.function = (void *)(fxgmac_phy_link_poll);
 #else
     init_timer_key(&pdata->expansion.phy_poll_tm, 0, "fuxi_phy_link_update_timer", NULL);
-#endif
     pdata->expansion.phy_poll_tm.expires = jiffies + HZ / 2;
     pdata->expansion.phy_poll_tm.function = (void *)(fxgmac_phy_link_poll);
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,15,0))
     pdata->expansion.phy_poll_tm.data = (unsigned long)pdata;
 #endif
     add_timer(&pdata->expansion.phy_poll_tm);
@@ -368,6 +373,10 @@
 
 void fxgmac_phy_timer_destroy(struct fxgmac_pdata *pdata)
 {
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6,15,0))
+    timer_delete_sync(&pdata->expansion.phy_poll_tm);
+#else
     del_timer_sync(&pdata->expansion.phy_poll_tm);
+#endif
     DPRINTK("fxgmac_phy_timer removed\n");
 }
